src/eos_gates.js

```diff
@@ -40,14 +42,62 @@ const KonamiManager = new Lang.Class({
 });
 Signals.addSignalMethods(KonamiManager.prototype);
 
+function recordMetrics(event, data) {
+    let recorder = EosMetrics.EventRecorder.get_default();
+    recorder.record_event(event, data);
+}
+
+function actionButtonProps(props, application) {
+    // We check for either no flatpak-info or not already having
+    // a replacement, since we need the flatpak info to install the app
+    // from the app store. If we already have a replacement, then it is
+    // guaranteed that we have some sort of option (either through
+    // the flatpak info or the desktop file) to launch an existing
+    // application.
+    if (!props.replacement || (!props.replacement.flatpakInfo && !props.alreadyHaveReplacement))
+        return {
+            label: _("OK"),
+            action: Lang.bind(application, application.quit)
+        };
+
+    let appName = !!props.replacement.replacementInfo ?
+                  props.replacement.replacementInfo.appName : props.replacement.appName;
+
+
+    if (props.alreadyHaveReplacement)
+        return {
+            label: _("Launch %s").format(appName),
+            action: function() {
+                launchReplacementApp(props.replacement,
+                                     props.attempt.argv);
+                application.quit();
+            }
+        };
+
+    return {
+        label: _("Install %s in App Store").format(appName),
```

Comment:
We call it `App Center`. However, I believe we could just use `Install %s` here.